scripts/vci-directory-auditor/src/index.ts

```diff
@@ -1,172 +1,255 @@
-    iss: string,
-    name: string,
-    canonical_iss?: string,
-    website?: string
-}
-
-// directory structure
-interface TrustedIssuers {
-    participating_issuers: TrustedIssuer[]
-}
-
-// issuer log info
-interface IssuerInfoWithKeys {
-    // the issuer info
-    issuer: TrustedIssuer,
-    // the issuer's JWK set
-    keys: JWK.Key[],
-}
-
-interface IssuerLogInfo extends IssuerInfoWithKeys {
-    // errors while retrieving the issuer JWK set, if any
-    errors?: string[]
-} 
-
-// Key identifiers (KID) of one issuer
-interface IssuerKids {
-    iss: string,
-    kids: string[]
-}
-
-// a snapshot of the directory, including keys
-interface DirectorySnapshot {
-    // directory URL
-    directory: string,
-    // snapshot time
-    time: string,
-    // directory issuers
-    issuerInfo: IssuerInfoWithKeys[]
-}
-
-// a snapshot of the directory, including keys
-interface DirectoryLog {
-    // directory URL
-    directory: string,
-    // snapshot time
-    time: string,
-    // directory issuers
-    issuerInfo: IssuerLogInfo[]
-}
-
+import Url from 'url-parse';
+import { AuditLog, CRL, DirectoryLog, IssuerKey, IssuerKids, IssuerLogInfo, TrustedIssuers } from './interfaces';
+import { auditTlsDetails, getDefaultTlsDetails } from './bcp195';
 
 interface KeySet {
-    keys : JWK.Key[]
+    keys : IssuerKey[]
 }
 
 interface Options {
+    outlog: string;
+    outsnapshot: string;
+    previous: string;
+    auditlog: string;
     dirpath: string;
-    outpath: string;
-    outlogpath: string;
+    verbose: boolean;
 }
 
 //
 // program options
 //
 const program = new Command();
-program.requiredOption('-d, --dirpath <dirpath>', 'path of the directory to audit');
-program.option('-o, --outpath <outpath>', 'output path for the directory with keys');
-program.option('-l, --outlogpath <outlogpath>', 'output path for the directory log');
+program.option('-o, --outlog <outlog>', 'output directory log storing directory issuer keys, TLS details, CRLs, and errors/warnings');
+program.option('-s, --outsnapshot <outsnapshot>', 'output snapshot file storing directory issuer keys for non-erroneous issuers');
+program.option('-p, --previous <previous>', 'directory log file from a previous audit, for comparison with current one');
+program.option('-a, --auditlog <auditlog>', 'output audit file on the directory');
+program.option('-d, --dirpath <dirpath>', 'path of the directory to audit');
+program.option('-v, --verbose', 'verbose mode');
 program.parse(process.argv);
 const currentTime = new Date();
 
 // process options
 const options = program.opts() as Options;
-if (!options.outpath) {
-    options.outpath = path.join(DEFAULT_LOG_LOCATION, `directory_snapshot_${date.format(currentTime, 'YYYY-MM-DD-HHmmss')}.json`);
+if (!options.dirpath) {
+    options.dirpath = "../../vci-issuers.json";
+}
+if (!options.outlog) {
+    options.outlog = path.join('logs', `directory_log_${date.format(currentTime, 'YYYY-MM-DD-HHmmss')}.json`);
```

Comment:
Yes, this will need to be created when automation is setup. This is what I did in [my project](https://github.com/christianpaquin/vci-directory-auditor/).